packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -22,27 +23,100 @@ function exists(filename) {
 }
 
 export default function buildConfigChain(opts: Object = {}) {
-  const filename = opts.filename;
-  const builder = new ConfigChainBuilder();
-
-  builder.mergeConfig({
-    options: opts,
-    alias: "base",
-    dirname: process.cwd(),
-  });
-
-  // resolve all .babelrc files
-  if (opts.babelrc !== false) {
-    builder.findConfigs(filename);
+  const filename = opts.filename ? path.resolve(opts.filename) : null;
+  const builder = new ConfigChainBuilder(filename);
+
+  try {
+    builder.mergeConfig({
+      options: opts,
+      alias: "base",
+      dirname: process.cwd(),
+    });
+
+    // resolve all .babelrc files
+    if (opts.babelrc !== false) {
+      builder.findConfigs(filename);
+    }
+  } catch (e) {
+    if (e.code !== "BABEL_IGNORED_FILE") throw e;
```

Comment:
I don't _love_ this so if people want me to revisit I can, it's just the bailing out of a deep merge is kind of a pain with the current setup unfortunately.